tools/mo/openvino/tools/mo/ops/transpose.py

```diff
@@ -49,4 +51,16 @@ def infer(node):
         else:
             node.out_port(0).data.set_shape(input_shape[order])
 
+    @staticmethod
+    def reverse_infer(node: Node):
+        input_shape = node.in_port(0).data.get_shape()
```

Comment:
I actually do reverse_infer only if input_shape is None.
https://github.com/openvinotoolkit/openvino/blob/6ab34578feea9af43df825a638fdcd9da24fb22e/tools/mo/openvino/tools/mo/ops/transpose.py#L56-L61

I prefer to keep in this Transpose case one `if` instead of having multiple  `if`'s and `returns`. 
This looks shorter:
```
input_shape = node.in_port(0).data.get_shape()
output_shape = node.out_port(0).data.get_shape()
order = node.in_port(1).data.get_value()

if input_shape is None and output_shape is not None and order is not None:
    ...
```
than this:
```
if input_shape is None and output_shape is not None and order is not None:
    ...
input_shape = node.in_port(0).data.get_shape()
if input_shape is None:
    return
output_shape = node.out_port(0).data.get_shape()
if input_shape is None:
    return
order = node.in_port(1).data.get_value()
if input_shape is None:
    return
```